app/assets/javascripts/components/course_creator/course_creator.jsx

```diff
@@ -65,7 +65,12 @@ const CourseCreator = React.createClass({
 
   storeDidChange() {
     this.setState(getState());
-    this.state.tempCourseId = CourseUtils.generateTempId(this.state.course);
+    if (this.state.course.term !== '' && this.state.course.school !== '' && this.state.title !== '') {
```

Comment:
I think this function should just be in charge of returning empty string if either title or school is missing, and course_utils.js should be updated to not include the _() if term is empty.